baseplate/integration/__init__.py

```diff
@@ -5,6 +5,39 @@
 
 """
 from .wrapped_context import WrappedRequestContext
+
+
+class _LazyAttributesMixin(object):
+    """A mixin that allows instances to add a lazily evaluated attribute
+       through `add_lazy_property`.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(_LazyAttributesMixin, self).__init__(*args, **kwargs)
+        self._lazy_attributes = {}
+
+    # Called when an attribute is missing
+    def __getattr__(self, name):
+        try:
+            lazy_fn = self._lazy_attributes[name]
+        except KeyError:
+            return super(_LazyAttributesMixin, self).__getattr__(name)
+
+        prop_value = lazy_fn(self)
+        setattr(self, name, prop_value)
```

Comment:
Does this actually _do_ anything?  Isn't `__getattr__` always going to check `self._lazy_attributes[name]` first and find the function in there?  I think we would have to `pop` the function out of `self._lazy_attributes` for this to work.

Reply:
Side note, I think this would make the performance much worse since you're re-building the client object each time you access it.

Reply:
`__getattr__` is only called for things not in the instance's `__dict__` (to override _that_, you use `__getattribute__`) so it will stop getting called once the setattr puts the reified value into the instance dict.

Reply:
ahhhh, that's not confusing at all.......